src/conn.c

```diff
@@ -33,30 +34,114 @@ typedef struct conn_mode_entry {
 	int (*get_addrs_func)(juice_agent_t *agent, addr_record_t *records, size_t size);
 
 	mutex_t mutex;
-	conn_registry_t *registry;
+	conn_registry_t **registries;
+	int registries_size;
+	int registries_count;
```

Comment:
This is smart but there is a catch: this will result in a different registry, and therefore a different thread, to be created for every port. For instance it is a normal use case to create peerconnections on different ports in POLL mode (to set up per-container redirections for instance), and with this modification it will result in a new thread for each connection.

The mux listener is really a mux thing, that's why I wanted to move it to the mux implementation in https://github.com/paullouisageneau/libjuice/pull/248. I think there should be multiple callbacks for the mux registry instead of multiple registries.

Reply:
Maybe I'm missing something, but the `registry_impl` struct [holds the socket reference](https://github.com/paullouisageneau/libjuice/blob/2fb91a3c90b9e5bde2f3cb0ecb4f53b9af027eca/src/conn_mux.c#L37) so it would need multiples of this (along with multiple `send_mutex`s, and other properties) to listen on multiple ports?

I've almost refactored this locally, but I can't see how I'd get the local host/port combo in `lookup_agent` to find the appropriate callback [here](https://github.com/paullouisageneau/libjuice/blob/master/src/conn_mux.c#L300).  I have a reference to the registry, but the registry only listens on a single port.

Reply:
Yes, my bad, I meant multiple sockets for the mux registry, making it closer to the poll mode.

Reply:
Thinking about it, a simpler way would be to make getting the registry dependent of the registry type, and only create multiple registries in mux mode.

Reply:
Ok to confirm so I don't implement the wrong thing:

- Multiple registries in mux mode only
- Each registry contains a socket

Presumably each registry still has to have it's own thread to read from that socket, otherwise it'd need to switch to poll mode, which we don't want?

FWIW I don't see multiple sockets being used in production much, mostly just for testing purposes and even then not more than a handful so multiple threads might not be so bad.